autopilot/stim/sound/jackclient.py

```diff
@@ -271,33 +351,77 @@ def process(self, frames):
                         self.logger.exception(f'Continuous mode was set but got exception with continuous queue:\n{e}')
                         data = self.zero_arr
 
-                    self.client.outports[0].get_array()[:] = data.T
-
                 else:
-                    for channel, port in zip(self.zero_arr.T, self.client.outports):
-                        port.get_array()[:] = channel
+                    # Play zeros
+                    data = np.zeros(self.blocksize, dtype='float32')
+                
+                # Write data
+                self.write_to_outports(data)
+                
                 # sound is over
                 self.play_evt.clear()
                 self.stop_evt.set()
+                
             else:
-                if data.shape[0] < self.blocksize:
-                    # if sound was not padded, fill remaining with continuous sound or silence
-                    n_from_end = self.blocksize - data.shape[0]
-                    if self.continuous.is_set():
-                        # data = np.concatenate((data, self.continuous_cycle.next()[-n_from_end:]),
-                        #                       axis=0)
-                        try:
-                            cont_data = next(self.continuous_cycle)
-                            data = np.concatenate((data, cont_data[-n_from_end:]),
-                                                  axis=0)
-                        except Exception as e:
-                            self.logger.exception(f'Continuous mode was set but got exception with continuous queue:\n{e}')
-                            data = np.pad(data, (0, n_from_end), 'constant')
-                    else:
-                        data = np.pad(data, (0, n_from_end), 'constant')
-
-                self.client.outports[0].get_array()[:] = data.T
-
-
-
+                ## There is data available
+                # Write
+                self.write_to_outports(data)
+    
+    def write_to_outports(self, data):
```

Comment:
yep that method absolutely needs to be broken up, i like this. and thanks for commenting ^^